internal/installation/app/app_darwin.go

```diff
@@ -0,0 +1,244 @@
+package app
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ActiveState/cli/internal/assets"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/installation"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/osutils/user"
+	"github.com/ActiveState/cli/internal/strutils"
+)
+
+const (
+	execFileSource       = "exec.sh.tpl"
+	launchFileSource     = "com.activestate.platform.app.plist.tpl"
+	launchFileFormatName = "com.activestate.platform.%s.plist"
+	autostartFileSource  = "com.activestate.platform.autostart.plist.tpl"
+	iconFile             = "icon.icns"
+)
+
+var targetDirs = []string{
+	"/Contents",
+	"/Contents/MacOS",
+	"/Contents/Resources",
+}
+
+func (a *App) install() error {
+	// Create all of the necessary directories and files in a temporary directory
+	// Then move the temporary directory to the final location which for macOS will be the Applications directory
+	tmpDir, err := ioutil.TempDir("", "state-svc-")
```

Comment:
`state-svc-` seems to be assuming we are setting up a svc app, but at this level we should not have any knowledge about what app we are configuring. Maybe use a.Name.